libraries/ms-drivers/src/adt7476a_fan_controller.c

```diff
@@ -27,36 +78,53 @@ StatusCode adt7476a_set_speed(I2CPort port, uint8_t speed_percent, AdtPwmPort pw
 
     status_ok_or_return(i2c_write(port, adt7476a_i2c_address, adt7476a_speed_register,
                                   SIZEOF_ARRAY(adt7476a_speed_register)));
+    #ifdef MPXE
+      s_speed[0] = real_speed;
+    #endif
 
   } else if (pwm_port == ADT_PWM_PORT_2) {
     uint8_t adt7476a_speed_register[] = { ADT7476A_PWM_3, real_speed };
 
     status_ok_or_return(i2c_write(port, adt7476a_i2c_address, adt7476a_speed_register,
                                   SIZEOF_ARRAY(adt7476a_speed_register)));
+    #ifdef MPXE
+      s_speed[1] = real_speed;
+    #endif
 
   } else if (pwm_port == ADT_PWM_PORT_3) {
     return STATUS_CODE_UNIMPLEMENTED;
   } else {
     return STATUS_CODE_INVALID_ARGS;
   }
 
+  #ifdef MPXE
+    prv_export_store();
+  #endif
   return STATUS_CODE_OK;
 }
 
 StatusCode adt7476a_get_status(I2CPort port, uint8_t adt7476a_i2c_read_address,
                                uint8_t *register_1_data, uint8_t *register_2_data) {
   // read interrupt status register
-  status_ok_or_return(i2c_read_reg(port, adt7476a_i2c_read_address,
-                                   ADT7476A_INTERRUPT_STATUS_REGISTER_1, register_1_data,
-                                   ADT7476A_REG_SIZE));
-  status_ok_or_return(i2c_read_reg(port, adt7476a_i2c_read_address,
-                                   ADT7476A_INTERRUPT_STATUS_REGISTER_2, register_2_data,
-                                   ADT7476A_REG_SIZE));
-
-  return STATUS_CODE_OK;
+  StatusCode reg1 =
+      (i2c_read_reg(port, adt7476a_i2c_read_address, ADT7476A_INTERRUPT_STATUS_REGISTER_1,
+                    register_1_data, ADT7476A_REG_SIZE));
+  StatusCode reg2 =
+      (i2c_read_reg(port, adt7476a_i2c_read_address, ADT7476A_INTERRUPT_STATUS_REGISTER_2,
+                    register_2_data, ADT7476A_REG_SIZE));
+  #ifdef MPXE
+    s_status[0] = reg1 ? true : false;
+    s_status[1] = reg2 ? true : false;
+
+    prv_export_store();
+  #endif
+  return (reg1 || reg2) ? STATUS_CODE_UNKNOWN : STATUS_CODE_OK;
```

Comment:
I think this can remain as STATUS_CODE_OK, since the point of the return here is to indicate whether or not we successfully got the status, not whether or not the status is ok

Reply:
I got rid of the `status_ok_or_return` function. Will that still return if it fails then?

Reply:
i2c_read will never fail on x86 so we don't need to worry there